src/components/Icon/Icon.component.tsx

```diff
@@ -0,0 +1,36 @@
+import React, {createElement} from 'react';
+
+import {IconProps} from './Icon';
+import IconMap from './Icon.map';
+import {StyledIconContainer} from './Icon.styles';
+
+/**
+ * Icon Component
+ *
+ * @example Correct usage with default size and color
+ * ```ts
+ * <Icon name="Unlink" />
+ * ```
+ *
+ * @example Using custom size and color
+ * ```ts
+ * <Icon name="Unlink" size="36" iconColor="#000" />
+ * ```
+ */
+export const Icon = (props: IconProps) => {
```

Comment:
can we destructure the props directly here?

Reply:
We can, but I think we should not expose props in the functional component's argument. I have destructured props directly in the function's argument for sub-components.